hazelcast/src/test/java/com/hazelcast/cluster/TcpIpSplitBrainDiscoveryTest.java

```diff
@@ -80,15 +81,16 @@ public void cleanup() {
     }
 
     @Test
+    @RepeatOnNodeFailure
     public void testSplitBrainRecoveryFromInitialSplit() throws IOException {
         instances.add(Hazelcast.newHazelcastInstance(createConfigWithRestEnabled(5801, 5901)));
         instances.add(Hazelcast.newHazelcastInstance(createConfigWithRestEnabled(5901, 5801)));
-        assertClusterSizeEventually(2, Arrays.asList(instances.get(0), instances.get(1)), 10);
-
+        assertClusterSizeEventually(2, instances.subList(0, 2), 10);
 
         instances.add(Hazelcast.newHazelcastInstance(createConfigWithRestEnabled(6001, 6101)));
         instances.add(Hazelcast.newHazelcastInstance(createConfigWithRestEnabled(6101, 6001)));
-        assertClusterSizeEventually(2, Arrays.asList(instances.get(2), instances.get(3)), 10);
+        assertClusterSizeEventually(2, instances.subList(2, 4), 10);
+
         updateMemberList();
         assertClusterSizeEventually(4, instances, 10);
```

Comment:
This is the assertion that failed:

```
    <failure message="Cluster size is not correct. Expected: 4, actual: 3, instance index: 0" type="java.lang.AssertionError">java.lang.AssertionError: Cluster size is not correct. Expected: 4, actual: 3, instance index: 0
        at org.junit.Assert.fail(Assert.java:89)
        at com.hazelcast.test.HazelcastTestSupport.assertClusterSize(HazelcastTestSupport.java:1032)
        at com.hazelcast.test.HazelcastTestSupport.lambda$assertClusterSizeEventually$9(HazelcastTestSupport.java:1067)
        at com.hazelcast.test.HazelcastTestSupport.assertTrueEventually(HazelcastTestSupport.java:1247)
        at com.hazelcast.test.HazelcastTestSupport.assertTrueEventually(HazelcastTestSupport.java:1264)
        at com.hazelcast.test.HazelcastTestSupport.assertClusterSizeEventually(HazelcastTestSupport.java:1067)
        at com.hazelcast.test.HazelcastTestSupport.assertClusterSizeEventually(HazelcastTestSupport.java:1055)
        at com.hazelcast.cluster.TcpIpSplitBrainDiscoveryTest.testSplitBrainRecoveryFromInitialSplit(TcpIpSplitBrainDiscoveryTest.java:93)
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:90)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:55)
        at java.lang.reflect.Method.invoke(Method.java:508)
        at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
        at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
        at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56)
        at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
        at com.hazelcast.test.FailOnTimeoutStatement$CallableStatement.call(FailOnTimeoutStatement.java:115)
        at com.hazelcast.test.FailOnTimeoutStatement$CallableStatement.call(FailOnTimeoutStatement.java:107)
        at java.util.concurrent.FutureTask.run(FutureTask.java:277)
        at java.lang.Thread.run(Thread.java:825)
</failure>
```

Reply:
I agree that rerunning flaky tests is not preferable, but there is no other solution I can think of. Due to the limitations of the test environment (I guess), some nodes might be terminated. In that case, the tests that specifically checks the cluster topology are guaranteed to fail. I just thought instead of doing `run-lab-run` manually, we can let the test runner to rerun the affected tests.

Reply:
Since `HazelcastInstanceNotActiveException` is not guaranteed to throw in case of node failures, I introduced `assertNoFailedInstances(Collection<HazelcastInstance>)` method. You can check for failures at appropriate points for the sake of coverage and early-exit.